onnxruntime/core/codegen/common/utils.h

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+#include <cassert>
+#include <iomanip>
+#include <iostream>
+#include <memory>
+#include <string>
+#include <vector>
+
+namespace onnxruntime {
+
+// Holding utility functions that are not tied to TVM and ORT
+
+std::unique_ptr<char[]> GetEnv(const char* var);
+
+// Check if an environment variable is set
+bool IsEnvVarDefined(const char* var);
+
+int64_t TotalSize(const std::vector<int64_t>& shape);
+
+template <typename T1, typename T2>
+void DumpArrayRecursive(const T1* data, int64_t& data_offset, const std::vector<T2>& shape, int idx) {
```

Comment:
>DumpArrayRecursive [](start = 5, length = 18)

it looks like more suitable to put in debug folder.

Reply:
putting it in debug_dump.h/cc

---
In reply to: [294436241](https://github.com/microsoft/onnxruntime/pull/888#discussion_r294436241) [](ancestors = 294436241)